kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/jms/BlockingResponseCallback.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.client.jms;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+import org.kie.server.api.marshalling.Marshaller;
+import org.kie.server.api.model.ServiceResponse;
+import org.kie.server.api.model.ServiceResponsesList;
+import org.kie.server.api.model.Wrapped;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Simple blocking response callback backed by blocking queue that will allow sequential access to
+ * responses and block
+ * <ul>
+ *     <li>client if there is no message yet</li>
+ *     <li>server if the queue is full</li>
+ * </ul>.
+ */
+public class BlockingResponseCallback implements ResponseCallback {
+
+    private static final Logger logger = LoggerFactory.getLogger(BlockingResponseCallback.class);
+    private BlockingQueue<ServiceResponsesList> responses = new ArrayBlockingQueue<ServiceResponsesList>(10);
```

Comment:
What is the purpose of this `BlockingQueue` with a capacity of 10? Is `BlockingResponseCallback` intended to be used for retrieving only a single response and this queue capacity is just a "random" number? Or do we expect this callback to be reused for multiple requests so the queue will contain multiple responses? If the latter is true, I would recommend to add another constructor with this signature: `public BlockingResponseCallback(Marshaller marshaller, int queueCapacity)` Because it does not make sense for users to implement their own `ResponseCallback` just to be able to handle more than 10 responses.